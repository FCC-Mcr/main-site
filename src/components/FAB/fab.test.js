import React from "react"
import { render, fireEvent, getByAltText } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import FabMenu from "./"

describe("Fab Menu", () => {
  test("Initially renders with only menu button", () => {
    const { container, getByAltText } = render(<FabMenu />)
    expect(container.childElementCount).toEqual(1)
    expect(getByAltText("menu button")).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test("OnClick renders fab nav menu", () => {
    const { container, getByAltText } = render(<FabMenu />)
    fireEvent.click(getByAltText("menu button"))
    expect(container.childElementCount).toEqual(2)
    expect(container.querySelector("nav")).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test("Once opened re-clicking the menu button will close the menu", () => {
    const { container, getByAltText } = render(<FabMenu />)

    // click to open then close
    fireEvent.click(getByAltText("menu button"))
    fireEvent.click(getByAltText("menu button"))

    expect(container.childElementCount).toEqual(1)
    expect(getByAltText("menu button")).toBeInTheDocument()
    expect(container.childElementCount).toEqual(1)
    expect(container.querySelector("nav")).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
