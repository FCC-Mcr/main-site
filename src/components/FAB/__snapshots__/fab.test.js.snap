// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Menu Initially renders with only menu button 1`] = `
<div>
  <div
    class="fab u-shadow--3"
  >
    <img
      alt="menu button"
      src="[object Object]"
    />
  </div>
</div>
`;

exports[`Fab Menu OnClick renders fab nav menu 1`] = `
<div>
  <div
    class="fab u-shadow--3"
  >
    <img
      alt="menu button"
      src="[object Object]"
    />
  </div>
  <nav
    class="menu"
  >
    <a
      aria-current="page"
      class=""
      href="/"
    >
      Home
    </a>
    <a
      href="/blog/"
    >
      Blog
    </a>
    <a
      href="/learning-resources/"
    >
      Resources
    </a>
    <a
      href="/upcoming-talks/"
    >
      Upcoming Talks
    </a>
  </nav>
</div>
`;

exports[`Fab Menu Once opened re-clicking the menu button will close the menu 1`] = `
<div>
  <div
    class="fab u-shadow--3"
  >
    <img
      alt="menu button"
      src="[object Object]"
    />
  </div>
</div>
`;
